api/src/main/java/io/strimzi/api/kafka/model/listener/KafkaExternalListenerConfiguration.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.listener;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import io.strimzi.api.kafka.model.CertAndKeySecretSource;
+import io.strimzi.api.kafka.model.UnknownPropertyPreserving;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+/**
+ * Configures External listeners
+ */
+
+@JsonSubTypes({
+        @JsonSubTypes.Type(name = KafkaListenerExternalRoute.TYPE_ROUTE, value = RouteListenerConfiguration.class),
+        @JsonSubTypes.Type(name = KafkaListenerExternalLoadBalancer.TYPE_LOADBALANCER, value = LoadBalancerListenerConfiguration.class),
+        @JsonSubTypes.Type(name = KafkaListenerExternalNodePort.TYPE_NODEPORT, value = NodePortListenerConfiguration.class),
+        @JsonSubTypes.Type(name = KafkaListenerExternalIngress.TYPE_INGRESS, value = IngressListenerConfiguration.class),
+})
```

Comment:
The above classes are not subtypes of `KafkaExternalListenerConfiguration`. Why are they declared here?

Reply:
Oh yeah, I had some trouble with generating CRD. It should not be here.